### PR TITLE
1369 comment delete

### PR DESCRIPTION
**Pull request by sblack-usu**
#1369 
image

### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [x] Peer Code review and approval

### Positive Test Case
1. Login as an admin, find a comment and confirm you can delete the comment.
2. Login as a resource owner, find a comment on your resource not created by you and confirm you can delete the comment.
3. Login as a normal user, make a comment and confirm you can delete the comment.
4. Delete a comment with a comment thread.  Confirm the confirmation message includes messaging about the threaded comments.  When you delete the comment, the thread should not appear either.

### CODE REVIEW

File: theme/views.py [hunk 1]
```diff
@@ -565,6 +567,18 @@ def email_verify_password_reset(request, uidb36=None, token=None):
         messages.error(request, _("The link you clicked is no longer valid."))
         return redirect("/")
 
+@login_required()
+def delete_resource_comment(request, id):
+    comment = get_object_or_404(Comment, id=id)
+    comment.content_object.raccess.owners.filter(pk=request.user.pk).exists()
```
Comment (hyi) [inline theme/views.py:573]: Should Line 573 be deleted?

File: theme/templates/generic/includes/comment.html [hunk 2]
```diff
@@ -16,6 +17,13 @@ <h4>{{ comment.user|contact }}
                 <a href="#reply-{{ comment.id }}" class="reply">
                     <span class="glyphicon glyphicon-share-alt"></span> {% trans "Reply" %}
                 </a>
+                {%  if user.is_authenticated %}
+                    {% if comment.user == user or user.is_staff or self_access_level == 'owner' %}
```
Comment (hyi) [inline theme/templates/generic/includes/comment.html:21]: I see is_staff is used in the template but is_superuser is used in the view function. Should they be made consistent?

File: hs_core/models.py [hunk 3]
```diff
@@ -313,7 +313,9 @@ def page_permissions_page_processor(request, page):
     else:
         is_version_of = ''
 
-    permissions_allow_copy = request.user.uaccess.can_view_resource(cm)
+    permissions_allow_copy = False
+    if request.user.is_authenticated:
+        permissions_allow_copy = request.user.uaccess.can_view_resource(cm)
```
Comment (hyi) [inline hs_core/models.py:318]: can_view_resources will return False is the user is not authenticated or not able to view this resource, right? If that is the case, lines 316 and 317 are not really needed, right?

Comment (hyi) [reply hs_core/models.py:318]: OK. Never mind. I got it. Anonymous users need to be handled explicitly here.

File: theme/static/js/generic-resource.js [hunk 4]
```diff
@@ -150,6 +150,23 @@ function removeExtraMetaTable(table) {
...
+    else{
+        $("#delete_comment_message").text("This will delete this comment")
+    }
+    $('#deleteCommentDialog').modal('show');
+}
+
+function removeComment() {
+    $("#deleteCommentDialog").modal('hide');
+    window.location = "/comment/delete/" + $("#delete_comment_id").val().trim();
+}
```
Comment (hyi) [inline theme/static/js/generic-resource.js:168]: Is this removeComment() function needed? I see the view function return a http redirect to refresh the whole resource page, so why not trigger the view function url directly in the modal dialog? Are you able to directly use #delete_comment_id value in the delete comment modal dialog?

File: theme/views.py [hunk 5]
```diff
@@ -1,5 +1,7 @@
 from json import dumps
 
+from django_comments.models import Comment
+from django_comments.views.moderation import perform_delete
```
Comment (hyi) [inline theme/views.py:4]: Nice to be able to leverage perform_delete function directly from django_comments module.

File: theme/static/js/generic-resource.js (hunk 4 above)
Comment (sblack-usu) [reply theme/static/js/generic-resource.js:168]: I followed the patterns set by the other modal confirmation boxes.  I think there are probably easier patterns out there, but I think we should stay consistent.

File: theme/templates/generic/includes/comment.html (hunk 2 above)
Comment (sblack-usu) [reply theme/templates/generic/includes/comment.html:21]: well, probably.  As far as I can tell, they resolve the same way, it's just is_staff seems to be the mezzanine term for is_superuser?  This template was already using is_staff, so I went with it here.  I'll try updating the endpoint to check for is_staff.

File: theme/views.py (hunk 1 above)
Comment (sblack-usu) [reply theme/views.py:573]: yes it should

**Comment by hyi** [comment]
@sblack-usu I googled and found is_staff is more limited than is_superuser. From https://stackoverflow.com/questions/20469174/whats-the-difference-between-staff-admin-superuser-in-django, it seems to indicate is_staff gives a user permission to log into admin interface, but may not have permissions unless a superuser assigns permission to staff users. I am thinking this may apply to use cases that a super user gives a staff user limited permission to, for example, only adding or deleting users for user management, but no permission to do other stuff. So I think in our case, using is_superuser may be more appropriate than is_staff for comment deletion. If you agree, you can commit another minor change to use is_superuser. Otherwise, I can just approve this PR since for our uses, we pretty much treat is_staff as a superuser, so they both should work to serve our purpose for now.

**Comment by sblack-usu** [comment]
@hyi - I updated it to user is_superuser

**Pull request merged**